### PR TITLE
Add AMQP extension

### LINKED ISSUE #143: Create a Camel AMQP extension

### PR DESCRIPTION
**Pull request by jamesnetherton**
Fixes #143

### CODE REVIEW

File: extensions/amqp/pom.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                <scope>import</scope>
+            </dependency>
+            <!-- quarkus-qpid-jms-deployment is currently not part quarkus-qpid-jms-deployment-->
+            <dependency>
+                <groupId>org.amqphub.quarkus</groupId>
+                <artifactId>quarkus-qpid-jms-deployment</artifactId>
+                <version>${quarkus.qpid.jms.version}</version>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
```
Comment (ppalaga) [inline extensions/amqp/pom.xml]: I'd expect these to be in the deployment module. Is there any reason for them to be here in the parent?

Comment (ppalaga) [reply extensions/amqp/pom.xml]: By importing the deployment BOMs here in the parent, we awake an impression that it is legal to use the deployment artifacts in all descendants, incl. the runtime module, which I think really should not happen.

Comment (jamesnetherton) [reply extensions/amqp/pom.xml]: I'll fix this.

I also noticed some other extensions where this happens. I'll follow up with another PR to fix tomorrow.

Comment (ppalaga) [reply extensions/amqp/pom.xml]: I have some dep. management cleanup in progress. I'd fix all these things there, if you do not mind.

**Pull request merged**